scripts/kubestellar

```diff
@@ -0,0 +1,221 @@
+#!/usr/bin/env bash
+# Copyright 2023 The KCP Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# Usage: $0 (start or stop | -v)
+
+# Purpose: control whether the controllers are running, setting up the ESPW if necessary.. The following components are created:
+#           (a) 1 kcp workspace: edge service provider workspace (espw)
+#           (b) 3 kubestellar controllers: kubestellar-scheduler, mailbox-controller and placement-translator
+
+# Assumption: kcp server is running.
+
+# Requirements:
+#    Download KubeStellar binaries
+#    KubeStellar controller binaries are on $PATH.
+
+
+set -e
+
+install=""
+verbosity=0
+remove=0
+cleanup=0
+espw_name="espw"
+log_folder=$(pwd)/kubestellar-logs
+
+while (( $# > 0 )); do
+    case "$1" in
+    (start)
+        install=1;; 
+    (stop) 
+        install=0;; 
+    (--log-folder)
+        if (( $# > 1 ));
+        then { log_folder="$2"; shift; }
+        else { echo "$0: missing log folder" >&2; exit 1; }
+        fi;;
+    (--verbose|-V)
+        verbosity=1;;
+    (-h|--help)
+        echo "Usage: $0 [start| stop][--log-folder log_folder] [-V|--verbose]"
+        exit 0;;
+    (-*)
+        echo "$0: unknown flag" >&2 ; exit 1;
+        exit 1;;
+    (*)
+        echo "$0: unknown positional argument" >&2; exit 1;
+        exit 1;;
+    esac
+    shift
+done
+
+
+# Check if docker is running
+if ! docker ps > /dev/null
+then
+  echo "Docker Not running ...."
+  exit 1
+fi
+
+# Check go version
+go_version=`go version | { read _ _ v _; echo ${v#go}; }`
+
+function ver { printf "%03d%03d%03d%03d" $(echo "$1" | tr '.' ' '); }
+if [ $(ver $go_version) -lt $(ver 1.19) ]; then
+    echo "Update your go version to at least 1.19"
+fi
+
+# Check if a given process name is running
+process_running() {
+  SERVICE="$1"
+  if pgrep -f "$SERVICE" >/dev/null
+  then
+      echo "running"
+  else
+      echo "stopped" 
+  fi
+}
+
+# Check if kcp is running
+if [ $(process_running kcp) != "running" ]
+then
+    echo "kcp is not running - please start it ...."
+    exit 1
+fi
+
+# Check mailbox-controller is already running
+if [ $(process_running mailbox-controller) == "running" ]
+then
+    echo "An older deployment of mailbox-controller is already running - terminating it ...."
+    pkill -f mailbox-controller
```

Comment:
I think it would be better if this script did not do unnecessary restarts of controllers.  Unnecessary restarts lead to unnecessary loss of log information (at least in the current way of launching), which will make our life hard as soon as users start reporting problems.

Reply:
this script works, and that is all we need for now.  Please push this change to another PR and we can debate it after we publish the first version of our quickstart.